src/mapeo-manager.js

```diff
@@ -640,38 +640,44 @@ export class MapeoManager extends TypedEmitter {
 
     // Any errors from here we need to remove project from db because it has not
     // been fully added and synced
+    let project = null
     try {
       // 4. Write device info into project
-      const project = await this.getProject(projectPublicId)
-
-      try {
-        const deviceInfo = this.getDeviceInfo()
-        if (hasSavedDeviceInfo(deviceInfo)) {
-          await project[kSetOwnDeviceInfo](deviceInfo)
-        }
-      } catch (e) {
-        // Can ignore an error trying to write device info
-        this.#l.log(
-          'ERROR: failed to write project %h deviceInfo %o',
-          projectKey,
-          e
-        )
-      }
-
-      // 5. Wait for initial project sync
-      if (waitForSync) {
-        await this.#waitForInitialSync(project)
-      }
-
-      this.#activeProjects.set(projectPublicId, project)
+      project = await this.getProject(projectPublicId)
     } catch (e) {
+      // Only happens if getProject fails
       this.#l.log('ERROR: could not add project', e)
       this.#db
         .delete(projectKeysTable)
         .where(eq(projectKeysTable.projectId, projectId))
         .run()
       throw e
     }
+
+    try {
+      const deviceInfo = this.getDeviceInfo()
+      if (hasSavedDeviceInfo(deviceInfo)) {
+        await project[kSetOwnDeviceInfo](deviceInfo)
+      }
+    } catch (e) {
+      // Can ignore an error trying to write device info
+      this.#l.log(
+        'ERROR: failed to write project %h deviceInfo %o',
+        projectKey,
+        e
+      )
+    }
+
+    this.#activeProjects.set(projectPublicId, project)
+
+    // 5. Wait for initial project sync
+    if (waitForSync) {
+      try {
+        await this.#waitForInitialSync(project)
+      } catch (e) {
+        this.#l.log('ERROR: could not do initial project sync', e)
```

Comment:
We maybe do want to throw here, but with an error that can be correctly handled in the front-end, e.g. to explain that this is a "non-fatal" error that can be recovered by subsequently syncing. However for this initial iteration I would maybe leave this as a silent error, since it will require additional frontend work to handle the error.